### PR TITLE
Search: Combine other filters into filter__ep_do_intercept_request

### PR DESCRIPTION
**Pull request by rebeccahum**
## Description
Inspired by #2703, let's re-factor the functionality of `get_cached_index_exists_request` and `invalidate_cached_index_exists_request` into one filter: `filter__ep_do_intercept_request`. This way we can ensure no requests are made when concurrency is hit.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] I've created a changelog description that aligns with the provided examples.

## Steps to Test
Test to ensure that index_exists caching and invalidation of it is still functional:

1) Shell in
2) Make an index exists request:
```
$indexable = \ElasticPress\Indexables::factory()->get( 'post' );
$indexable->index_exists();
```
3) get the option to ensure it's cached: `get_option( 'es_index_exists_vip-200508-post-1' );`
4) delete index `wp vip-search delete-index`
5) repeat `#3` but ensure it returns false

### CODE REVIEW

**Comment by sjinks** [comment]
Awesome! I'll take a closer look tomorrow morning

File: search/includes/classes/class-search.php [hunk 1]
```diff
@@ -916,6 +879,10 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 			// Return a generic VIP Search WP_Error instead of the one from wp_remote_request
 			return new \WP_Error( 'vip-search-upstream-request-failed', 'There was an error connecting to the upstream search server' );
 		} else {
```
Comment (sjinks) [inline search/includes/classes/class-search.php:881]: We can probably drop this `else` statement, as the `if` branch ends with `return`.

**Review by sjinks — DISMISSED** [review]

**Comment by sjinks** [comment]
Nice! When you merge this into master, I will rebase #2703 

Thank you!

File: search/includes/classes/class-search.php (hunk 1 above)
Comment (sjinks) [reply search/includes/classes/class-search.php]: No longer relevant.

**Review by sjinks — APPROVED** [review]

**Comment by rebeccahum** [comment]
r2104-stacks

**Comment by rebeccahum** [comment]
v20211215.0

**Pull request merged**